### PR TITLE
 Addressed PA-82937 validation of delete shared backups

### PR DESCRIPTION
**Pull request by apimpalgaonkar**
**What this PR does / why we need it**

1. Share multiple backups with new user
2. delete those shared backups from new user
3. validate that backups have been deleted successfully from the owner side as well

**Which issue(s) this PR fixes (optional)**
https://portworx.testrail.net/index.php?/cases/view/82937

#Closes 
https://portworx.atlassian.net/browse/PA-518

Aetos link : http://aetos.pwx.purestorage.com/resultSet/testSetID/80018

### CODE REVIEW

**Review by snigdha-px — APPROVED** [review]
LGTM

File: tests/backup/backup_test.go [hunk 1]
```diff
@@ -2800,6 +2800,214 @@ var _ = Describe("{ShareBackupWithDifferentRoleUsers}", func() {
...
+	var backupLocationUID string
+	var cloudCredUID string
+	var CloudCredUidList []string
+	var appContexts []*scheduler.Context
+	var bkpNamespaces []string
+	var clusterUid string
+	var clusterStatus api.ClusterInfo_StatusInfo_Status
+	var credName string
+	bkpNamespaces = make([]string, 0)
+	providers := getProviders()
```
Comment (lsrinivas-pure) [inline tests/backup/backup_test.go:2825]: move these function call inside It

File: tests/backup/backup_test.go [hunk 2]
```diff
@@ -2800,6 +2800,214 @@ var _ = Describe("{ShareBackupWithDifferentRoleUsers}", func() {
...
+	})
+	It("Validate shared backups are deleted from owner of backup ", func() {
+		ctx, err := backup.GetAdminCtxFromSecret()
+		log.FailOnError(err, "Fetching px-central-admin ctx")
+		Step("Validate applications", func() {
+			log.InfoD("Validate applications")
+			ValidateApplications(contexts)
+		})
+		Step("Create Users", func() {
+			err = backup.AddUser(userName, firstName, lastName, email, password)
```
Comment (lsrinivas-pure) [inline tests/backup/backup_test.go:2851]: handle the err here

File: tests/backup/backup_test.go [hunk 3]
```diff
@@ -2800,6 +2800,214 @@ var _ = Describe("{ShareBackupWithDifferentRoleUsers}", func() {
...
+func GetAllBackupsAdmin() ([]string, error) {
+	var bkp *api.BackupObject
+	backupNames := make([]string, 0)
+	backupDriver := Inst().Backup
+	ctx, err := backup.GetAdminCtxFromSecret()
+	log.FailOnError(err, "Fetching ctx")
+
+	bkpEnumerateReq := &api.BackupEnumerateRequest{
+		OrgId: orgID}
+	curBackups, err := backupDriver.EnumerateBackup(ctx, bkpEnumerateReq)
```
Comment (lsrinivas-pure) [inline tests/backup/backup_test.go:3004]: handl err here

**Review by lsrinivas-pure — CHANGES_REQUESTED** [review]

File: tests/backup/backup_test.go (hunk 1 above)
Comment (apimpalgaonkar) [reply tests/backup/backup_test.go:2825]: Added these in It block

File: tests/backup/backup_test.go (hunk 2 above)
Comment (apimpalgaonkar) [reply tests/backup/backup_test.go:2851]: err handled

File: tests/backup/backup_test.go (hunk 3 above)
Comment (apimpalgaonkar) [reply tests/backup/backup_test.go:3004]: err handled

**Comment by apimpalgaonkar** [comment]
DashBoard link (new run) - http://aetos.pwx.purestorage.com/resultSet/testSetID/80724

**Review by lsrinivas-pure — APPROVED** [review]
LGTM

**Review by kshithijiyer-px — APPROVED** [review]

**Pull request merged**